### PR TITLE
Introduce Pico ConfigBean Builder Extensions

### PR DESCRIPTION
**Pull request by trentjeff**
For #5443 .

Note that there are some todo items left like validation. It definitely needs a draft review before I go much further though.

### CODE REVIEW

File: builder/builder/src/main/java/io/helidon/builder/RequiredAttributeVisitor.java [hunk 1]
```diff
@@ -38,23 +38,42 @@
...
+     *
+     * @param allowNullsByDefault true if nulls should be allowed
+     */
+    // important note: this needs to remain public since it will be new'ed from code-generated builder processing ...
+    public RequiredAttributeVisitor(boolean allowNullsByDefault) {
+        this.allowNullsByDefault = allowNullsByDefault;
     }
 
     @Override
+    // important note: this needs to remain public since it will be new'ed from code-generated builder processing ...
```
Comment (tomas-langer) [inline builder/builder/src/main/java/io/helidon/builder/RequiredAttributeVisitor.java:62]: Probably wrong copy/paste?

File: builder/builder/src/main/java/io/helidon/builder/RequiredAttributeVisitor.java [hunk 2]
```diff
@@ -71,6 +90,7 @@ public void visit(String attrName,
      *
      * @throws java.lang.IllegalStateException when any attributes are in violation with the validation policy
      */
+    // important note: this needs to remain public since it will be new'ed from code-generated builder processing ...
```
Comment (tomas-langer) [inline builder/builder/src/main/java/io/helidon/builder/RequiredAttributeVisitor.java:93]: The comment is probably wrong here? `new'ed` does not make sense in this context...

**Review by tomas-langer — APPROVED** [review]

File: builder/builder/src/main/java/io/helidon/builder/RequiredAttributeVisitor.java (hunk 1 above)
Comment (trentjeff) [reply builder/builder/src/main/java/io/helidon/builder/RequiredAttributeVisitor.java:62]: done in https://github.com/helidon-io/helidon/pull/5591/files

File: builder/builder/src/main/java/io/helidon/builder/RequiredAttributeVisitor.java (hunk 2 above)
Comment (trentjeff) [reply builder/builder/src/main/java/io/helidon/builder/RequiredAttributeVisitor.java:93]: done in https://github.com/helidon-io/helidon/pull/5591/files

**Pull request merged**